<head>
  <meta charset="utf-8">
  <title>bulk RNAseq</title>

  <meta name="generator" content="Hugo 0.55.3" />

  <!-- mobile responsive meta -->
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  
  <!-- ** Plugins Needed for the Project ** -->
  <!-- Bootstrap -->
  <link rel="stylesheet" href="https://pilm-bioinformatics.github.io/knowledgebase/plugins/bootstrap/bootstrap.min.css">
  <!-- themefy-icon -->
  <link rel="stylesheet" href="https://pilm-bioinformatics.github.io/knowledgebase/plugins/themify-icons/themify-icons.css">
  <!-- highlight -->
  <link rel="stylesheet" href="https://pilm-bioinformatics.github.io/knowledgebase/plugins/highlight/hybrid.css">
  <!-- fonts -->
  <link href="https://fonts.googleapis.com/css?family=Lato:300,400,700" rel="stylesheet">

  <style>
  :root{
    --primary-color:#004b7e;
    --secondary-color:#f9f9f9;
    --text-color:#636363;
    --text-color-dark:#242738;
    --white-color:#ffffff;
  }  
  </style>

  <!-- Main Stylesheet -->
  
  <link rel="stylesheet" href="https://pilm-bioinformatics.github.io/knowledgebase/css/style.min.css" integrity="" media="screen">

  <!-- jquiry -->
  <script src="https://pilm-bioinformatics.github.io/knowledgebase/plugins/jquery/jquery-1.12.4.js"></script>

  <!-- jquary ui -->
  <script src="https://pilm-bioinformatics.github.io/knowledgebase/plugins/jquery/jquery-ui.js"></script>

  <!--Favicon-->
  <link rel="icon" href="https://pilm-bioinformatics.github.io/knowledgebase/images/favicon.png" type="image/x-icon">

</head>

<!-- navigation -->
<header class="shadow-bottom sticky-top bg-white">
  <div class="container">
    <nav class="navbar navbar-expand-lg navbar-light">
  <a class="navbar-brand" href="https://pilm-bioinformatics.github.io/knowledgebase/">PILM-MIT</a>
  <button class="navbar-toggler border-0" type="button" data-toggle="collapse" data-target="#navigation"
    aria-controls="navigation" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>

  <div class="collapse navbar-collapse text-center" id="navigation">
    <ul class="navbar-nav ml-auto">
      <li class="nav-item">
        <a class="nav-link text-dark" href="https://pilm-bioinformatics.github.io/knowledgebase/">Home</a>
      </li>
      
      
      <li class="nav-item">
          <a class="nav-link text-dark" href="/knowledgebase/faq">Faq</a>
      </li>
      
      
      
      <li class="nav-item">
          <a class="nav-link text-dark" href="https://github.mit.edu/pages/PILM-bioinformatics/knowledgebase/">Private-Space</a>
      </li>
      
      
      
      <li class="nav-item dropdown">
        <a class="nav-link dropdown-toggle text-dark" href="#" role="button" data-toggle="dropdown" aria-haspopup="true"
          aria-expanded="false">
          Documents
        </a>
        <div class="dropdown-menu" >
          
          <a class="dropdown-item" href="/knowledgebase/environment">Environment</a>
          
          <a class="dropdown-item" href="/knowledgebase/analysis">Analysis Type</a>
          
          <a class="dropdown-item" href="/knowledgebase/resources">Resources</a>
          
          <a class="dropdown-item" href="https://github.mit.edu/pages/PILM-bioinformatics/knowledgebase/resources/">Private Resources</a>
          
          <a class="dropdown-item" href="https://github.mit.edu/pages/PILM-bioinformatics/knowledgebase/admin/">Core Organization</a>
          
        </div>
      </li>
      
      
      
      <li class="nav-item">
          <a class="nav-link text-dark" href="https://github.mit.edu/pages/PILM-bioinformatics/secure-newsletters">Newsletter</a>
      </li>
      
      
      
      <li class="nav-item">
          <a class="nav-link text-dark" href="https://github.mit.edu/PILM-bioinformatics/support/issues/new">contact</a>
      </li>
      
      
    </ul>
    
    <ul class="list-inline text-center mb-0">
      
    </ul>
  </div>
</nav>
  </div>
</header>
<!-- /navigation -->


<section class="single pt-5 bg-gray">
  <div class="container">
    <div class="row">
      <div class="col-lg-3">
        <div class="p-4 bg-white sticky-top top-100">
            
            <nav class="sidebar-menu">
              <ul class="list-styled">
            <li class=""><a href="/knowledgebase/resources/">Resources</a>
              <ul class="">
            <li class=""><a href="/knowledgebase/resources/diy/">DIY:Online tutorials</a>
            </li>
            <li class=""><a href="/knowledgebase/resources/diyours/">DIY:PILM tutorials</a>
            </li></ul>
              
            </li>
            <li class="parent d-block "><a href="/knowledgebase/analysis/">Analysis Type</a>
              <ul class="sub-menu">
            <li class=" active "><a href="/knowledgebase/analysis/rnaseq/">bulk RNAseq</a>
            </li></ul>
              
            </li>
            <li class=""><a href="/knowledgebase/environment/">System Environment</a>
              <ul class="">
            <li class=""><a href="/knowledgebase/environment/conda/">conda</a>
            </li></ul>
              
            </li>
            <li class=""><a href="/knowledgebase/faq/">Frequently Asked Questions</a>
            </li></ul>
            </nav>

            
        </div>
      </div>
      <div class="col-lg-9">
        <div class="p-5 bg-white">
            <h2>bulk RNAseq</h2>
            

<h3 id="fast-rnaseq-pipeline">Fast RNAseq pipeline</h3>

<p>Here we show how to run a NextFlow pipeline for RNAseq from species with well
annotated genes, like human or mouse.</p>

<h4 id="raw-data">Raw data</h4>

<p>This is an example using a public dataset.</p>

<p>We got the following links from <a href="https://ewels.github.io/sra-explorer/">sra-explore</a>, a web-based app
that help you to find public datasets and provides a script to download the data.</p>

<p>We use the following files:</p>

<pre><code class="language-bash">#!/usr/bin/env bash
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307897/SRR307897_1.fastq.gz -o SRR307897_GSM758559_CshlLong_RnaSeq_GM12878_cell_longPolyA_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307897/SRR307897_2.fastq.gz -o SRR307897_GSM758559_CshlLong_RnaSeq_GM12878_cell_longPolyA_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307912/SRR307912_1.fastq.gz -o SRR307912_GSM758566_CshlLong_RnaSeq_H1-hESC_cell_longPolyA_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307912/SRR307912_2.fastq.gz -o SRR307912_GSM758566_CshlLong_RnaSeq_H1-hESC_cell_longPolyA_2.fastq.gz
</code></pre>

<ul>
<li>create a folder <code>raw</code></li>
<li>type <code>cd raw</code></li>
<li>copy that into a file called <code>download.sh</code></li>
<li>type <code>bash download.sh</code>.</li>
</ul>

<p>Wait until is finished, then come back to parent folder <code>cd ..</code>.</p>

<h4 id="annotation">Annotation</h4>

<p>We need the transcriptome fasta file as well:</p>

<p><code>wget ftp://ftp.ensembl.org/pub/release-96/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz</code></p>

<h4 id="environment">Environment</h4>

<p>If you are in the PILM-cluster, follow this instructions: [PRIVATE PILM]</p>

<p>If you want to try from scratch, follow these <a href="/environment/conda">instructions</a> to install
miniconda, and then type:</p>

<ul>
<li><code>conda -n create -y nf-fastrnaseq nextflow salmon multiqc fastqc</code></li>
<li><code>conda activate nf-fastrnaseq</code></li>
</ul>

<h4 id="get-the-pipeline">Get the pipeline</h4>

<p>If you are in  the  PILM-cluster, follow this instructions: PRIVATE PILM</p>

<p>If you want to try from scratch, download the pipeline from GitHub: PUBLIC PILM.</p>

<p>Now that you know where the file <code>paired.nf</code> is, you can run the following command to get the pipeline working:</p>

<p><code>nextflow run paired.nf -w working -name tiny -with-report all-report -with-timeline all-time --outdir results  -qs 6 --reads &quot;raw/*{1,2}.fastq.gz&quot; --transcriptopme Homo_sapiens.GRCh38.cdna.all.fa.gz</code></p>

            <p class="post-meta border-bottom pb-3 mb-0">Updated on 29 Dec 2018</p><nav class="pagination mt-3"><a class="nav nav-prev" href="/knowledgebase/analysis/" title="Analysis Type"><i class="ti-arrow-left mr-2"></i>Analysis Type</a>
            <a class="nav nav-next" href="/knowledgebase/environment/" title="System Environment">System Environment<i class="ti-arrow-right ml-2"></i></a>
            </nav></div>
      </div>
    </div>
  </div>
</section>


<!-- footer -->
<footer class="section bg-gray pb-0">
  <div class="container">
    <div class="row">
      <div class="col-md-8 text-md-left text-center">
        <p>The Hugo Documents are copyright © gethugothemes 2019 - Content © PILM.</p> 
      </div>
      <div class="col-md-4 text-md-right text-center">
        <ul class="list-inline mb-3">
          
          <li class="list-inline-item"><a class="text-color d-inline-block p-2" href="https://twitter.com/lopantano"><i class="ti-twitter-alt"></i></a></li>
          
        </ul>
      </div>
    </div>
  </div>
</footer>
<!-- /footer -->

<!-- Bootstrap JS -->
<script src="https://pilm-bioinformatics.github.io/knowledgebase/plugins/bootstrap/bootstrap.min.js"></script>
<!-- highlight -->
<script src="https://pilm-bioinformatics.github.io/knowledgebase/plugins/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<!-- Main Script -->

<script src="https://pilm-bioinformatics.github.io/knowledgebase/js/script.min.js"></script>